ZenPacks/zenoss/OpenStackInfrastructure/events.py

```diff
@@ -19,11 +19,38 @@
                                                            get_port_fixedips,
                                                            is_uuid,
                                                            )
+from Products.ZenModel.ZVersion import VERSION as ZENOSS_VERSION
+from distutils.version import LooseVersion
+
 import ast
 
 import logging
 LOG = logging.getLogger('zen.OpenStack.events')
 
+# Starting from zenoss 6.4.0 ADM code were reworked without full compatibility 
+# with lefacy directives provided by PythonCollector zenpack, add some tweaks here 
+# to make directives work with old and new ADM
+NEW_ADM = '6.4.0'
+
+
+def addUpdateDirective(objmap):
+    # Set directive correctly based on platform ADM code.
+    if LooseVersion(ZENOSS_VERSION) >= LooseVersion(NEW_ADM):
+        objmap._update = True
+    else:
+        objmap._add = False
+    return objmap
+
+def addRemoveDirective(objmap):
+    # remove update directive if we add new one as it won't work correctly
+    # with a new ADM code
+    if LooseVersion(ZENOSS_VERSION) >= LooseVersion(NEW_ADM):
+        if hasattr(objmap, '_update'):
+            del objmap._update
+
+    objmap._remove = True
+    return objmap
+
```

Comment:
I have an alternate suggestion (given below) but I should mention that this code could be written so that the version test is made only once.  Why test for a result that'll never change in the current process?

Alternative implementation:
```
if LooseVersion(ZENOSS_VERSION) >= LooseVersion(NEW_ADM):

    def addUpdateDirective(objmap):
        objmap._update = True
        return objmap

    def addRemoveDirective(objmap):
        if hasattr(objmap, "_update"):
            del objmap._update
        objmap._remove = True
        return objmap

else:

    def addUpdateDirective(objmap):
        objmap._add = False
        return objmap

    def addRemoveDirective(objmap):
        objmap._remove = True
        return objmap
```